File: rgctl-tests/ecommerce-csharp/src/Ecommerce/Models/CartItem.cs
namespace Ecommerce.Models;

public class CartItem
{
    public long Id { get; set; }
    public long CartId { get; set; }
    public long ProductId { get; set; }
    public int Quantity { get; set; }

    public Cart Cart { get; set; } = null!;
    public Product Product { get; set; } = null!;
}

File: rgctl-tests/ecommerce-c/src/models/cart_item.c
#include "ecommerce/models/cart_item.h"
#include <string.h>

void cart_item_init(cart_item_t *obj) { if (obj) { memset(obj, 0, sizeof(*obj)); } }

double cart_item_total_price(const cart_item_t *item, double unit_price) { return item ? item->quantity * unit_price : 0; }

